device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransportMessage.java

```diff
@@ -202,14 +202,11 @@ public boolean isMessageAckNeeded(IotHubClientProtocol protocol)
     {
         if (protocol == IotHubClientProtocol.MQTT || protocol == IotHubClientProtocol.MQTT_WS)
         {
-            if (this.operationType == DeviceOperations.DEVICE_OPERATION_TWIN_SUBSCRIBE_DESIRED_PROPERTIES_REQUEST
-                    || this.operationType == DeviceOperations.DEVICE_OPERATION_METHOD_SUBSCRIBE_REQUEST
-                    || this.operationType == DeviceOperations.DEVICE_OPERATION_TWIN_UNSUBSCRIBE_DESIRED_PROPERTIES_REQUEST)
-            {
-                //This is a SUBSCRIBE action in MQTT which we currently treat synchronously, so there is no message ack
-                // unlike other send operations
-                return false;
-            }
+            //This is a SUBSCRIBE action in MQTT which we currently treat synchronously, so there is no message ack
+            // unlike other send operations
+            return this.operationType != DeviceOperations.DEVICE_OPERATION_TWIN_SUBSCRIBE_DESIRED_PROPERTIES_REQUEST
```

Comment:
Why not just keep the same logical OR? Is this recommended practice in java?
return x || y ||z vs return !x && !y && !z

Reply:
That's a very good question, I don't the answer. will do research and get back to you.